### DISCUSSION TITLE
How to obtain optimized length scale assigned to each dimension

### DISCUSSION THREAD

**Discussion by SwarnenduSekharGhosh** [SheffieldML/GPy#945] [Q&A] [answered]
If I am using automatic relevance determination inside RBF, when I print the model how can I get the optimized values of length scales assigned to each dimension? Also, if I use a combination of kernels then how to specifically obtain the length scales of the RBF kernel?

**Comment by lawrennd** [comment, marked as answer]
Hi Swarnendu, From memory it doesn't display because it's variable length and can be long if input dimension is high. 

Here's some code for displaying and changing values.

```
import GPy
import numpy as np
from IPython.display import display

kern = GPy.kern.RBF(4, ARD=True)
display(kern.lengthscale)

kern.lengthscale = [0.3, 1.2, 1.4, 1.2]
display(kern.lengthscale)
```

**Reply by SwarnenduSekharGhosh** [reply]
Thank you so much, sir. Appreciate your quick response.

**Reply by shadzzz90** [reply]
Hi @lawrennd, what if we are using different combination of kernels?
